contracts/staking/src/tests/ibc_transfer_tests.rs

```diff
@@ -0,0 +1,323 @@
+
+        let inflight_packet = INFLIGHT_PACKETS
+            .may_load(&deps.storage, sequence.clone())
+            .unwrap();
+        assert_eq!(Some(inflight_packet), Some(None));
+
+        let res = query_ibc_queue(deps.as_ref());
+        assert!(res.unwrap().ibc_queue.len() == 0);
+    }
+
+    #[test]
+    fn timeout_ibc_queue() {
+        let mut deps = init();
+        let env = mock_env();
+        let info = mock_info("creator", &coins(1000, NATIVE_TOKEN));
+        let msg = ExecuteMsg::LiquidStake {};
+        let res = execute(deps.as_mut(), mock_env(), info.clone(), msg.clone());
+        let ibc_sub_msg_id = env.block.time.nanos() + env.transaction.unwrap().index as u64;
+        match res {
+            Ok(ref result) => {
+                assert_eq!(
+                    result.attributes,
+                    vec![
+                        attr("action", "liquid_stake"),
+                        attr("sender", "creator"),
+                        attr("amount", "1000")
+                    ]
+                );
+            }
+            Err(e) => {
+                panic!("Unexpected error: {:?}", e);
+            }
+        }
+        let sequence = 1;
+
+        // Reply
+        let _result = reply(
+            deps.as_mut(),
+            mock_env(),
+            Reply {
+                id: ibc_sub_msg_id,
+                result: SubMsgResult::Ok(SubMsgResponse {
+                    data: Some(cosmwasm_std::Binary::from(MsgTransferResponse {
+                        sequence: sequence.clone(),
+                    })), // No data returned
+                    events: Vec::new(), // No events
+                }),
+            },
+        );
+
+        let _result = sudo(
+            deps.as_mut(),
+            mock_env(),
+            SudoMsg::IBCLifecycleComplete(IBCLifecycleComplete::IBCTimeout {
+                channel: CHANNEL_ID.to_string(),
+                sequence: sequence,
+            }),
+        );
+
+        let inflight_packet = INFLIGHT_PACKETS
+            .may_load(&deps.storage, sequence.clone())
+            .unwrap();
+        assert_eq!(
+            inflight_packet,
+            Some(ibc::IBCTransfer {
+                sequence: sequence.clone(),
+                amount: 1000,
+                status: ibc::PacketLifecycleStatus::TimedOut
+            })
+        );
+
+        let res = query_ibc_queue(deps.as_ref());
+        assert!(res.unwrap().ibc_queue.len() == 1);
+
+        // send recover message
+        let msg = ExecuteMsg::RecoverPendingIbcTransfers {};
+        let _res = execute(deps.as_mut(), mock_env(), info.clone(), msg);
+
+        let inflight_packet = INFLIGHT_PACKETS
+            .may_load(&deps.storage, sequence.clone())
+            .unwrap();
+        assert_eq!(Some(inflight_packet), Some(None));
+
+        let res = query_ibc_queue(deps.as_ref());
+        assert!(res.unwrap().ibc_queue.len() == 0);
```

Comment:
@faboweb could you please check it? I think `INFLIGHT_PACKETS` should be empty after recover() called